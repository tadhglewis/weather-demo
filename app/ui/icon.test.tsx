import React from "react";
import { render } from "@testing-library/react-native";
import Icon from "./icon";
import "jest-styled-components";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: "",
}));

describe("icon", () => {
  it("should render icon", () => {
    expect(render(<Icon icon="cloud" size={16} />)).toMatchSnapshot();
  });

  it("should render styled icon", () => {
    const tree = render(
      <Icon icon="font-awesome" size={32} style={{ backgroundColor: "red" }} />
    );

    expect(tree.toJSON()?.props.style.backgroundColor).toEqual("red");

    expect(tree).toMatchSnapshot();
  });
});
