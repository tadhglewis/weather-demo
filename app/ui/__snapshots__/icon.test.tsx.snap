// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon should render icon 1`] = `
<
  icon={
    Array [
      "fas",
      "cloud",
    ]
  }
  size={16}
/>
`;

exports[`icon should render styled icon 1`] = `
<
  icon={
    Array [
      "fas",
      "font-awesome",
    ]
  }
  size={32}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;
